app/src/utils/rateLimit/concurrencyRateLimits.ts

```diff
@@ -0,0 +1,52 @@
+import { sql } from "kysely";
+import { kysely } from "~/server/db";
+import { v4 as uuidv4 } from "uuid";
+import { RateLimitError } from "openai";
+
+export const recordOngoingRequestStart = async (projectId: string, proceed = true) => {
```

Comment:
Let's add some basic tests for this functionality, similar to the existing tests I have in rateLimits.test.ts

Reply:
Addressed all the other comments and tested manually. Will need some time to write tests.